fission/supporttool/resources/kubernetes.go

```diff
@@ -0,0 +1,260 @@
+	client   *kubernetes.Clientset
+	objType  string
+	selector string
+}
+
+func NewKubernetesObjectDumper(clientset *kubernetes.Clientset, objType string, selector string) Resource {
+	return KubernetesObjectDumper{
+		client:   clientset,
+		objType:  objType,
+		selector: selector,
+	}
+}
+
+func (res KubernetesObjectDumper) Dump(dumpDir string) {
+	switch res.objType {
+	case KubernetesService:
+		objs, err := res.client.CoreV1().Services(metav1.NamespaceAll).List(metav1.ListOptions{LabelSelector: res.selector})
+		if err != nil {
+			log.Printf("Error getting %v list with selector %v: %v", res.objType, res.selector, err)
+			return
+		}
+
+		for _, item := range objs.Items {
+			go func(obj corev1.Service) {
+				obj = serviceClean(obj)
+				f := getFileName(dumpDir, obj.ObjectMeta)
+				writeToFile(f, obj)
+			}(item)
+		}
+
+	case KubernetesDeployment:
+		objs, err := res.client.AppsV1().Deployments(metav1.NamespaceAll).List(metav1.ListOptions{LabelSelector: res.selector})
+		if err != nil {
+			log.Printf("Error getting %v list with selector %v: %v", res.objType, res.selector, err)
+			return
+		}
+
+		for _, item := range objs.Items {
+			go func(obj appsv1.Deployment) {
+				f := getFileName(dumpDir, obj.ObjectMeta)
+				writeToFile(f, obj)
+			}(item)
+		}
+
+	case KubernetesPod:
+		objs, err := res.client.CoreV1().Pods(metav1.NamespaceAll).List(metav1.ListOptions{LabelSelector: res.selector})
+		if err != nil {
+			log.Printf("Error getting %v list with selector %v: %v", res.objType, res.selector, err)
+			return
+		}
+
+		for _, item := range objs.Items {
+			go func(obj corev1.Pod) {
+				f := getFileName(dumpDir, obj.ObjectMeta)
+				writeToFile(f, obj)
+			}(item)
+		}
+
+	case KubernetesHPA:
+		objs, err := res.client.AutoscalingV2beta1().HorizontalPodAutoscalers(metav1.NamespaceAll).List(metav1.ListOptions{LabelSelector: res.selector})
+		if err != nil {
+			log.Printf("Error getting %v list with selector %v: %v", res.objType, res.selector, err)
+			return
+		}
+
+		for _, item := range objs.Items {
+			go func(obj autoscalingv2beta1.HorizontalPodAutoscaler) {
+				f := getFileName(dumpDir, obj.ObjectMeta)
+				writeToFile(f, obj)
+			}(item)
+		}
+
+	case KubernetesNode:
+		objs, err := res.client.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: res.selector})
+		if err != nil {
+			log.Printf("Error getting %v list with selector %v: %v", res.objType, res.selector, err)
+			return
+		}
+
+		for _, item := range objs.Items {
+			go func(obj corev1.Node) {
+				obj = nodeClean(obj)
+				// Node doesn't have namespace value, use name here
+				f := filepath.Clean(fmt.Sprintf("%v/%v", dumpDir, obj.Name))
+				getFileName(dumpDir, obj.ObjectMeta)
+				writeToFile(f, obj)
+			}(item)
+		}
+
+	default:
+		log.Printf("Unknown type: %v", res.objType)
+	}
+}
+
+// serviceClean remove sensitive data(e.g. public IP, external name) from service objects
+func serviceClean(svc corev1.Service) corev1.Service {
```

Comment:
You'll also have to clean  hostname/ip from svc.Status.LoadBalancerStatus or something like that...